src/test/java/co/rsk/federate/FederationWatcherTest.java

```diff
@@ -22,354 +22,342 @@
+    private static final FederationArgs SECOND_FEDERATION_ARGS = new FederationArgs(
+        SECOND_FEDERATION_MEMBERS, SECOND_FEDERATION_CREATION_TIME, CREATION_BLOCK_NUMBER, NETWORK_PARAMETERS);
+    private static final Federation SECOND_FEDERATION = FederationFactory.buildStandardMultiSigFederation(SECOND_FEDERATION_ARGS);
+
+    // Third federation constants
+    private static final List<FederationMember> THIRD_FEDERATION_MEMBERS =
+        getFederationMembersFromPksForBtc(5000, 6000, 7000);
+    private static final Instant THIRD_FEDERATION_CREATION_TIME = Instant.ofEpochMilli(7400L);
+    private static final FederationArgs THIRD_FEDERATION_ARGS = new FederationArgs(
+        THIRD_FEDERATION_MEMBERS, THIRD_FEDERATION_CREATION_TIME, CREATION_BLOCK_NUMBER, NETWORK_PARAMETERS);
+    private static final Federation THIRD_FEDERATION = FederationFactory.buildStandardMultiSigFederation(THIRD_FEDERATION_ARGS);
+
+    private final FederationProvider federationProvider = mock(FederationProvider.class);
+    private final Ethereum ethereum = mock(Ethereum.class);
+    private final FederationWatcher federationWatcher = new FederationWatcher(ethereum);
 
     @Test
-    void setsListenerUp() throws Exception {
-        Mockito.doAnswer((InvocationOnMock m) -> {
+    void whenFederationWatcherIsSetUp_shouldAddListener() throws Exception {
+        // Arrange
+        doAnswer((InvocationOnMock m) -> {
             Object listener = m.getArgument(0);
-            assertEquals("co.rsk.federate.FederationWatcher$FederationWatcherRskListener", listener.getClass().getName());
-            assertSame(TestUtils.getInternalState(watcher, "federationProvider"), federationProvider);
+            assertEquals(FederationWatcher.FederationWatcherRskListener.class, listener.getClass());
             return null;
-        }).when(ethereumMock).addListener(any());
+        }).when(ethereum).addListener(any());
 
-        watcher.setup(federationProvider);
-        verify(ethereumMock).addListener(any());
+        // Act
+        federationWatcher.setup(federationProvider);
+
+        // Assert
+        verify(ethereum).addListener(any());
+        assertSame(TestUtils.getInternalState(federationWatcher, "federationProvider"), federationProvider);
     }
 
     @Test
-    void triggersActiveFederationChange_none_to_active() throws Exception {
-        EthereumListenerAdapter rskListener = setupAndGetRskListener(Optional.empty(), Optional.empty());
-        class EventsLogger
-        {
-            public int activeCalls = 0;
-            public int retiringCalls = 0;
-        }
-
-        when(federationProvider.getActiveFederationAddress()).thenReturn(federation1.getAddress());
-        when(federationProvider.getActiveFederation()).thenReturn(federation1);
+    void whenNoActiveFederation_shouldTriggerActiveFederationChange() throws Exception {
+        // Arrange
+        var rskListener = setupAndGetRskListener(Optional.empty(), Optional.empty());
```

Comment:
why using `var` here too?

Reply:
It can help with readability a ton especially for tests :)